server/server.js

```diff
@@ -1,33 +1,95 @@
+import 'source-map-support/register';
 import express from 'express';
+import compression from 'compression';
+import morgan from 'morgan';
 import React from 'react';
 import { renderToString } from 'react-dom/server';
+import { matchPath } from 'react-router-dom';
+import StaticRouter from 'react-router-dom/StaticRouter';
+import ReactHelmet from 'react-helmet';
+import axios from 'axios';
+
 import App from '../common/App';
+import routes from '../common/routes';
 
-const app = express();
+const server = express();
 
-app.get('/api', (req, res) => {
-  res.send({
-    message: 'I am a server route and can also be hot reloaded!'
-  });
-});
+// Remove annoying Express header addition.
+server.disable('x-powered-by');
+
+// Compress (gzip) assets in production.
+server.use(compression());
+server.use(morgan(process.env.NODE_ENV === 'production' ? 'combined' : 'dev'));
+
+// Setup server side routing.
+server.get('*', (req, res) => {
+  const context = {};
+  const matches = routes.map(
+    // we'd probably want some recursion here so our routes could have
+    // child routes like `{ path, component, routes: [ { route, route } ] }`
+    // and then reduce to the entire branch of matched routes, but for
+    // illustrative purposes, sticking to a flat route config
+    (route, index) => {
+      const match = matchPath(req.url, route.path, route);
```

Comment:
Hey, I've been following this branch and I think there is a problem on this line:

The syntax for matchPath is incorrect:
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/matchPath.md

There is no third argument and the second argument should be an object which has path property?

This works fine:
const match = matchPath(req.url, route);